contracts/carbon/Strategies.sol

```diff
@@ -481,7 +482,8 @@ abstract contract Strategies is Initializable {
             params.tokens.source,
             params.trader,
             totals.sourceAmount,
-            params.txValue
+            params.txValue,
+            false
```

Comment:
Technically speaking, some sophisticated fee mechanism could detect that the function is called from create or update strategy, and avoid a fee only in those cases (which means that the fee will go undetected when trading).

In the previous comment, I've implied that we can do this check only when creating a strategy, in order to reduce strategy update gas cost.

In this comment, I'm implying that we should do it when creating a strategy, when updating a strategy and when trading, in order to ensure 100% no fee.

...

Reply:
I've also mentioned this - for me it makes sense to have it as well, since some tokens may activate fees after a strategy has been created. The main issue here is the increased gas cost of trading - getting the balance of one token twice will cost 2100 (cold read) + 100 (hot read) = 2200 more gas.

Reply:
Plus the external contract function call, which is another 3000 as far as I remember.
And you call it twice (before and after).
So that's 10k for one token, and there are possibly two tokens.

Reply:
only a single token on trade but I'm reluctant to increase gas costs on trades for that

Reply:
We deposit one side, since it's a trade - so only up to one token per trade. Additionally, two external contract function calls cost about 2700 total: 2600 (cold access, base) for the first access + 100 for the second (warm access).
So total cost should be 2200 for sload + ~2700 for external call = ~5000